proteus/mprans/RANS3PSed2D.h

```diff
@@ -828,41 +826,71 @@ namespace proteus
 					   double dmom_v_source[nSpace],
 					   double dmom_w_source[nSpace])
     {
-      double rho, mu,nu,H_mu,uc,duc_du,duc_dv,duc_dw,viscosity,H_s;
+      double rhoFluid, muFluid,nuFluid,H_mu,uc,duc_du,duc_dv,duc_dw,viscosity,H_s;
       H_mu = (1.0-useVF)*smoothedHeaviside(eps_mu,phi)+useVF*fmin(1.0,fmax(0.0,vf));
-      nu  = nu_0*(1.0-H_mu)+nu_1*H_mu;
-      rho  = rho_0*(1.0-H_mu)+rho_1*H_mu;
-      mu  = rho_0*nu_0*(1.0-H_mu)+rho_1*nu_1*H_mu;
-      viscosity = nu;//mu; gco check
+      nuFluid  = nu_0*(1.0-H_mu)+nu_1*H_mu;
+      rhoFluid  = rho_0*(1.0-H_mu)+rho_1*H_mu;
+      muFluid  = rho_0*nu_0*(1.0-H_mu)+rho_1*nu_1*H_mu;
+      //gco kinematic viscosity used, in sedclosure betaterm is multiplied by fluidDensity
+      viscosity = nuFluid;//mu; gco check
+      //phi_s is sediment fraction in this case - gco check
       uc = sqrt(u*u+v*v*+w*w); 
       duc_du = u/(uc+1.0e-12);
       duc_dv = v/(uc+1.0e-12);
       duc_dw = w/(uc+1.0e-12);
       double solid_velocity[3]={uStar,vStar,wStar}, fluid_velocity[3]={u_f,v_f,w_f};
-      double new_beta = closure.betaCoeff(1.0-phi_s,
-                                          rho,
+      double new_beta = closure.betaCoeff(phi_s,
+                                          rhoFluid,
                                           fluid_velocity,
                                           solid_velocity,
                                           viscosity);
-      //new_beta/=rho;
+      //new_beta/=rhoFluid;
       //std::cout<<"total "<<(1.0-phi_s)*new_beta<<std::endl;
-      mom_u_source += (1.0 - phi_s)*new_beta*(u-u_f);
-      mom_v_source += (1.0 - phi_s)*new_beta*(v-v_f);
+      mom_u_source += (phi_s)*new_beta*(u-u_f);
+      mom_v_source += (phi_s)*new_beta*(v-v_f);
       /* mom_w_source += phi_s*new_beta*(w-w_s); */
 
-      dmom_u_source[0] = (1.0 - phi_s)*new_beta;
+      dmom_u_source[0] = (phi_s)*new_beta;
       dmom_u_source[1] = 0.0;
       /* dmom_u_source[2] = 0.0; */
       
       dmom_v_source[0] = 0.0;
-      dmom_v_source[1] = (1.0 - phi_s)*new_beta;
+      dmom_v_source[1] = (phi_s)*new_beta;
       /*dmom_v_source[2] = 0.0;*/
 
       /*      dmom_w_source[0] = 0.0;
       dmom_w_source[1] = 0.0;
-      dmom_w_source[2] = (1.0 - phi_s)*new_beta;*/
+      dmom_w_source[2] = (phi_s)*new_beta;*/
     }
 
+    inline
+        void updateParticlePressure(const double phi_s,
```

Comment:
@adimako @cekees Could you check this part about frictional pressure in sediment phase?
What I mean is: should I add this term like it is?
The gradient is not needed because then we integrate through the cells, is it?

Reply:
Giovanni looks fine. 
I would remove the d_mom_source arrays completely. I do not think that there is a case where p_friction depends on the velocities